src/compiler/interface.jl

```diff
@@ -171,6 +186,41 @@ function copy!(x::AbstractVector,  gs::Grads)
   x
 end
 
+broadcasted(f, gss::Grads...) = map(f, gss...)
+
+for op in (:*, :/)
+  @eval broadcasted(::typeof($op), a::Number, gs::Grads) = map(x -> $op(a, x), gs)
+  @eval broadcasted(::typeof($op), gs::Grads, a::Number) = map(x -> $op(x, a), gs)
+end
+
+function materialize!(gs1::Grads, gs2::Grads)
```

Comment:
Why are we overloading this function? Broadcasting on dictionaries is reserved so messing with it seems odd

cc @oxinabox

Reply:
Grads is not a dictionary

Reply:
This seems fine and good to me.
Grads isn't a dictionary.
I also say dictionaries should get with it and define broadcast, there  are about two options and they just need to pick one.

It makes a lot of sense for Grads objects to be broadcast assignable inplace.
Which is what this function does, no? It makes `x .= y` work.

Means we can do gradient clipping via 
```julia
generalized_clamp(x::Number, low, high) = clamp(x, low, high)
generalized_clamp(x, low, high) = generalized_clamp.(x, low, high)

g .= generalized_clamp.(g, -300, 300)
```

This function is noncontentioius. Other things might be. but `materialize` isn't

Reply:
And in-place is wanted for the [original issue](https://github.com/FluxML/Flux.jl/issues/707) where you need to accumulate many gradients. In-place would let you do it without allocating too many copies of the model.